File: solutions/python/matching-brackets/1/matching_brackets.py
def is_paired(cadena):
    stack = []
    pares = {'(': ')', '[': ']', '{': '}'}

    for caracter in cadena:
        if caracter in pares:
            stack.append(caracter)
        elif caracter in pares.values():
            if not stack or pares[stack.pop()] != caracter:
                return False

    return not stack

File: solutions/javascript/matching-brackets/1/matching-brackets.js
//
// This is only a SKELETON file for the 'Matching Brackets' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

// []{}()

export const isPaired = (code) => {
  const pares = { "[": "]", "{": "}", "(": ")" };
  let codeClean = [...code.replace(/[^(){}[\]]/g, "").split("")];
  const acum = [];
  for (let simbol of codeClean) {
    if ("([{".includes(simbol)) {
      acum.push(pares[simbol]);
    } else {
      if (simbol !== acum.pop()) return false;
    }
  }
  if (acum.length > 0) return false;
  return true;
};
